cms/djangoapps/contentstore/views/tests/test_group_configurations.py

```diff
@@ -422,12 +592,144 @@ def test_cannot_delete_non_existent_group_configuration(self):
     """
-    Tests for usage information of configurations.
+    Tests for usage information of configurations and content groups.
     """
 
     def setUp(self):
         super(GroupConfigurationsUsageInfoTestCase, self).setUp()
 
+    def test_content_group_not_used(self):
+        """
+        Test that right data structure will be created if content group is not used.
+        """
+        self._add_user_partitions(scheme_id='cohort')
+        actual = GroupConfiguration.get_or_create_content_group_configuration_with_usage(self.store, self.course)
+        expected = {
+            'id': 0,
+            'name': 'Name 0',
+            'scheme': 'cohort',
+            'description': 'Description 0',
+            'version': UserPartition.VERSION,
+            'groups': [
+                {'id': 0, 'name': 'Group A', 'version': 1, 'usage': []},
+                {'id': 1, 'name': 'Group B', 'version': 1, 'usage': []},
+                {'id': 2, 'name': 'Group C', 'version': 1, 'usage': []},
+            ],
+        }
+        self.assertEqual(actual, expected)
+
+    def test_can_get_correct_usage_info_when_special_characters_are_in_content(self):
+        """
+        Test if content group json updated successfully with usage information.
+        """
+        self._add_user_partitions(count=1, scheme_id='cohort')
+        vertical, __ = self._create_problem_with_content_group(
+            cid=0, group_id=1, name_suffix='0', special_characters=u"JOSÉ ANDRÉS"
+        )
+
+        actual = GroupConfiguration.get_or_create_content_group_configuration_with_usage(self.store, self.course)
+        expected = {
+            'id': 0,
+            'name': "Name 0",
+            'scheme': 'cohort',
+            'description': 'Description 0',
+            'version': UserPartition.VERSION,
+            'groups': [
+                {
+                    'id': 0, 'name': 'Group A', 'version': 1, 'usage': []
+                },
+                {
+                    'id': 1, 'name': 'Group B', 'version': 1,
+                    'usage': [
+                        {
+                            'url': u"/container/{}".format(vertical.location),
+                            'label': u"Test Unit 0 / Test Problem 0JOSÉ ANDRÉS"
+                        }
+                    ]
+                },
+                {
+                    'id': 2, 'name': 'Group C', 'version': 1, 'usage': []
+                },
+            ],
+        }
+        self.assertEqual(actual, expected)
+
+    def test_can_get_correct_usage_info_for_content_groups(self):
+        """
+        Test if content group json updated successfully with usage information.
+        """
+        self._add_user_partitions(count=1, scheme_id='cohort')
+        vertical, __ = self._create_problem_with_content_group(cid=0, group_id=1, name_suffix='0')
+
+        actual = GroupConfiguration.get_or_create_content_group_configuration_with_usage(self.store, self.course)
+
+        expected = {
```

Comment:
This JSON is duplicated with only minor changes in three tests. I'd create a helper method that parameterizes just the differences. IMO that would make the test logic easier to follow.